domain/ide/workspace/workspace.go

```diff
@@ -174,10 +174,9 @@ func (w *Workspace) ChangeWorkspaceFolders(ctx context.Context, params lsp.DidCh
 	for _, folder := range params.Event.Added {
 		f := NewFolder(uri.PathFromUri(folder.Uri), folder.Name, w.scanner, w.hoverService, w.scanNotifier, w.notifier)
 		w.AddFolder(f)
-	}
-
-	if config.CurrentConfig().IsAutoScanEnabled() {
-		w.ScanWorkspace(ctx)
+		if config.CurrentConfig().IsAutoScanEnabled() {
+			f.ScanFolder(ctx)
+		}
```

Comment:
Adding a workspace folder should not trigger a workspace scan. It should just trigger a scan of the added folder.